### ISSUE TITLE
Feature Request: modify actions

### ISSUE THREAD

**Issue by rowrowrowrow** [aohua/redux-state-sync#90]
Hi, thanks for the great package!

Would you consider adding a function to the configuration for modifying the actions before they are synced?

**Comment by aohua** [comment]
Yea, sure, but could you provide a use case?

**Comment by rowrowrowrow** [comment]
Of course, I'd like to modify portions of state depending on whether broadcastchannel is available and don't have a feasible way to separate those pieces otherwise. I.e. if there will be storage limitations that prevent the sending and receiving via the fallbacks for broadcastchannel.

**Comment by rowrowrowrow** [comment]
@aohua I see a few issues that might cause someone to want to modify an action before it's shared. Namely general performance, storage limitations when broadcast channel isn't available, and the structured clone algorithm causing errors if some part of an action is not clonable.

**Comment by rowrowrowrow** [comment]
@aohua bump
